client/dashboard/me/monetize-subscriptions/monetize-subscription.tsx

```diff
@@ -0,0 +1,310 @@
+										actions: [
+											{
+												url: CALYPSO_CONTACT,
+												label: __( 'Please contact support' ),
+											},
+										],
+									} );
+								} else {
+									createErrorNotice(
+										__( 'There was a problem while stopping your subscription.' ),
+										{
+											actions: [
+												{
+													url: CALYPSO_CONTACT,
+													label: __( 'Please contact support' ),
+												},
+											],
+										}
+									);
+								}
+							},
+						} );
+					} }
+				>
+					{ title }
+				</Button>
+			}
+		/>
+	);
+}
+
+const BackButton = () => {
+	const router = useRouter();
+
+	return (
+		<Button
+			className="dashboard-page-header__back-button"
+			icon={ isRTL() ? chevronRight : chevronLeft }
+			onClick={ () => {
+				router.navigate( {
+					to: getMonetizeSubscriptionsUrl(),
+				} );
+			} }
+		>
+			{ getMonetizeSubscriptionsPageTitle() }
+		</Button>
+	);
+};
+
+export default function MonetizeSubscription() {
+	const params = monetizeSubscriptionRoute.useParams();
+	const subscriptionId: string = params.subscriptionId;
+
+	const { data: subscription } = useQuery( monetizeSubscriptionQuery( subscriptionId ) );
+
+	const { mutate: stopSubscription, isPending: isStoppingSubscription } = useMutation(
+		monetizeSubscriptionStop( subscriptionId )
+	);
+
+	const { mutate: disableAutoRenew, isPending: isDisablingAutoRenew } = useMutation(
+		monetizeSubscriptionDisableAutoRenew( subscriptionId )
+	);
+
+	const { mutate: enableAutoRenew, isPending: isEnablingAutoRenew } = useMutation(
+		monetizeSubscriptionResumeAutoRenew( subscriptionId )
+	);
+
+	const isRenewable = subscription && ( subscription.renew_interval || subscription.is_renewable ); // can remove renew_interval once backend is deployed
+	const isAutoRenewing = isRenewable && subscription.renew_interval;
+	const isProduct = !! subscription && ! isRenewable;
+	const isDisabledAutorenewing = isRenewable && ! subscription.renew_interval;
+	const isUpdating = isEnablingAutoRenew || isDisablingAutoRenew;
+
+	return (
+		<PageLayout
+			size="small"
+			header={
+				<PageHeader
+					prefix={ <BackButton /> }
+					title={ isProduct ? __( 'Product Details' ) : __( 'Subscription Details' ) }
+				/>
+			}
+		>
+			{ isStoppingSubscription && (
+				<Notice status="info">
+					{ isProduct ? __( 'Removing this product' ) : __( 'Stopping this subscription' ) }
+				</Notice>
+			) }
+			{ isUpdating && <Notice status="info">{ __( 'Updating subscription auto-renew' ) }</Notice> }
+			{ subscription && (
+				<Card>
+					<MonetizeSiteHeader name={ subscription.site_title } domain={ subscription.site_url } />
+
+					<CardHeader>
+						<HStack alignment="center">
+							<VStack>
+								<HStack alignment="center">
+									<Text weight={ 700 } size="title" as="h3">
+										{ subscription.title }
+									</Text>
+									<Text weight={ 700 }>
```

Comment:
This price looks a bit off, should it have the same size of the left text?

<img width="793" height="536" alt="Image" src="https://github.com/user-attachments/assets/bc39b591-02c7-449b-9e23-b4f66518146e" />

Reply:
I re-done all this view with components from the Billing views